polyA/calculate_score.py

```diff
@@ -70,5 +74,81 @@ def calculate_score(
                 chunk_score += gap_init
         else:
             chunk_score += sub_matrix[seq1_char + seq2_char]
-
+            if char_complexity_adjustment:
+                chunk_score += char_complexity_adjustment[seq2_char]
     return chunk_score
+
+
+def calc_target_char_counts(
+    query_seq: str,
+    target_seq: str,
+):
+    query_char_counts = {"A": 0, "C": 0, "G": 0, "T": 0}
+    other_chars = set()
+    total_chars = 0
+    for q_base, t_base in zip(query_seq, target_seq):
+        if t_base not in ["A", "G", "T", "C"]:
+            other_chars.add(t_base)
+            continue
+        if q_base == "-":
+            continue
+        query_char_counts[t_base] += 1
+        total_chars += 1
+    return query_char_counts, total_chars, other_chars
+
+
+def calculate_complexity_adjusted_score(
+    char_background_freqs: Optional[Dict[str, float]],
+    query_seq: str,
+    target_seq: str,
+    lamb: float,
+):
```

Comment:
Add a doc comment that explains this function in context. Also, include the actual computations being done. Pretend it's a math paper and you need to define, precisely, "complexity adjusted score". Feel free to use (pseudo) math notation even. This can help when debugging the function, it can also make it easier to write tests, since the definition is likely to be far more concise than the actual code since it can elide things like error handling.

Reply:
Also add some doc tests to this function and the one above.

Reply:
BTW there's a way to insert LaTeX style math notation into Python docstrings. See https://stackoverflow.com/questions/10081633/using-displaymath-directives-in-docstrings-formulas for an example.